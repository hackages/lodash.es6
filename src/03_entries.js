/**
 * Creates an array of own enumerable string keyed-value pairs for object
 */
export function entries() {}
