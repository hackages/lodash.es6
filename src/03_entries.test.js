import {entries} from './03_entries'

describe('entries', () => {
  describe('should pair values', () => {
    it('in a function', () => {
      function Test() {
        this.a = 32
        this.b = 45
      }

      expect(entries(new Test())).toMatchSnapshot()
    })

    it('in an object', () => {
      const Test = {
        a: 32,
        b: 45,
      }
      expect(entries(Test)).toMatchSnapshot()
    })

    it('in a class', () => {
      class Test {
        a = 32
        b = 45
      }
      expect(entries(new Test())).toMatchSnapshot()
    })
  })
})
