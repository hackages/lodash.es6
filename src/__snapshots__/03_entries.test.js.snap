// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entries should pair values in a class 1`] = `
Array [
  Array [
    "a",
    32,
  ],
  Array [
    "b",
    45,
  ],
]
`;

exports[`entries should pair values in a function 1`] = `
Array [
  Array [
    "a",
    32,
  ],
  Array [
    "b",
    45,
  ],
]
`;

exports[`entries should pair values in an object 1`] = `
Array [
  Array [
    "a",
    32,
  ],
  Array [
    "b",
    45,
  ],
]
`;
